koku/cost_models/serializers.py

```diff
@@ -183,8 +183,9 @@ class RateSerializer(serializers.Serializer):
 
     metric = serializers.DictField(required=True)
     cost_type = serializers.ChoiceField(choices=metric_constants.COST_TYPE_CHOICES)
+    description = serializers.CharField(allow_blank=True, max_length=100, required=False)
     tiered_rates = serializers.ListField(required=False)
-    tag_rates = serializers.ListField(required=False)
+    tag_rates = serializers.DictField(required=False)
```

Comment:
Hey @boaz0 after looking at the current UI prototype, I can see why you wanted to change the data structure here to a dictionary. However, I think leaving the data structure as a list provides us with more flexibility for how the tag rates can be used in the future. The dictionary is an acceptable data structure right now because every defined rate is being applied to anything that has that tag key and value. However, if we ever want the ability to decide tag pair A should be applied instead of tag pair B if both have been found, then we will probably need the data structure to be a list.

I think it's important to think about flexibility here because once we decide the data structure and this goes into production it's gonna be a bit more of a challenge to change in the future since it all lives in a json field in the db.

Reply:
No problem. I will revert this chane. Thanks for looking into it